packages/keplr/src/signer.ts

```diff
@@ -0,0 +1,246 @@
+import { AccountData, DirectSignResponse } from "@cosmjs/proto-signing";
+import {
+  ChainInfo,
+  Keplr,
+  Window as KeplrWindow,
+  KeplrSignOptions,
+} from "@keplr-wallet/types";
+import { SignDoc } from "cosmjs-types/cosmos/tx/v1beta1/tx";
+import { AminoSignResponse, StdSignDoc, Algo } from "@cosmjs/amino";
+import { assert } from "@cosmjs/utils";
+import { Signer, SignerStatus, SigningMode } from "@desmoslabs/desmjs";
+
+declare global {
+  interface Window extends KeplrWindow {}
+}
+
+export interface KeplrSignerOptions {
+  signingMode: SigningMode;
+  chainInfo: ChainInfo;
+  signOptions: KeplrSignOptions;
+}
+
+export const DesmosMainnet: ChainInfo = {
+  chainId: "desmos-mainnet",
+  chainName: "Desmos",
+  rpc: "https://rpc.mainnet.desmos.network",
+  rest: "https://api.mainnet.desmos.network",
+  bip44: {
+    coinType: 852,
+  },
+  bech32Config: {
+    bech32PrefixAccAddr: "desmos",
+    bech32PrefixAccPub: "desmospub",
+    bech32PrefixValAddr: "desmosvaloper",
+    bech32PrefixValPub: "desmosvaloperpub",
+    bech32PrefixConsAddr: "desmosvalcons",
+    bech32PrefixConsPub: "desmosvalconspub",
+  },
+  currencies: [
+    {
+      coinDenom: "DSM",
+      coinMinimalDenom: "udsm",
+      coinDecimals: 6,
+      coinGeckoId: "desmos",
+    },
+  ],
+  feeCurrencies: [
+    {
+      coinDenom: "udsm",
+      coinMinimalDenom: "udsm",
+      coinDecimals: 6,
+      coinGeckoId: "desmos",
+    },
+  ],
+  stakeCurrency: {
+    coinDenom: "DSM",
+    coinMinimalDenom: "udsm",
+    coinDecimals: 6,
+    coinGeckoId: "desmos",
+  },
+  gasPriceStep: {
+    low: 0.01,
+    average: 0.03,
+    high: 0.05,
+  },
+  features: ["stargate", "ibc-transfer", "no-legacy-stdTx", "ibc-go"],
+};
+
+export const DesmosTestnet: ChainInfo = {
+  chainId: "morpheus-apollo-2",
+  chainName: "Desmos Testnet",
+  rpc: "https://rpc.morpheus.desmos.network",
+  rest: "https://lcd.morpheus.desmos.network",
+  bip44: {
+    coinType: 852,
+  },
+  bech32Config: {
+    bech32PrefixAccAddr: "desmos",
+    bech32PrefixAccPub: "desmospub",
+    bech32PrefixValAddr: "desmosvaloper",
+    bech32PrefixValPub: "desmosvaloperpub",
+    bech32PrefixConsAddr: "desmosvalcons",
+    bech32PrefixConsPub: "desmosvalconspub",
+  },
+  currencies: [
+    {
+      coinDenom: "DARIC",
+      coinMinimalDenom: "udaric",
+      coinDecimals: 6,
+    },
+  ],
+  feeCurrencies: [
+    {
+      coinDenom: "udaric",
+      coinMinimalDenom: "udaric",
+      coinDecimals: 6,
+    },
+  ],
+  stakeCurrency: {
+    coinDenom: "DARIC",
+    coinMinimalDenom: "udaric",
+    coinDecimals: 6,
+  },
+  gasPriceStep: {
+    low: 0.01,
+    average: 0.03,
+    high: 0.05,
+  },
+  features: ["stargate", "ibc-transfer", "no-legacy-stdTx", "ibc-go"],
+};
+
+/**
+ * Signer that use Keplr to sign a transaction.
+ */
+export class KeplrSigner extends Signer {
+  public readonly chainInfo: ChainInfo = DesmosMainnet;
+
+  public readonly signingMode: SigningMode = SigningMode.DIRECT;
```

Comment:
Are there any instances where developers might want to access these field? Or can we make them private instead?

Reply:
They may want to know the active `chainInfo` to avoid misconfigurations and simplify the code  if they support hot-swap between mainet/testnet/custom, otherwise they would have to recreate the signer every time in order to ensure they use the right one or develop a custom (useless) logic.

Regarding the `signingMode`  i kept it as the `WalletConnect` signer (it might be useful to someone for the same reasons as `chainInfo`)
https://github.com/desmos-labs/desmjs/blob/c47678b0480b3727b1241f9120f840e548fad084/packages/walletconnect/src/signer.ts#L24

Reply:
> Regarding the `signingMode` i kept it as the `WalletConnect` signer (it might be useful to someone for the same reasons as `chainInfo`)
OK, perfect. Thanks for the reference.

> They may want to know the active `chainInfo` to avoid misconfigurations and simplify the code if they support hot-swap between mainet/testnet/custom, otherwise they would have to recreate the signer every time in order to ensure they use the right one or develop a custom (useless) logic.
Is it sufficient to leave the `chainInfo` field as `readonly` to support hotswap between different chains? Or should we also expose a `switchChain` method that allows changing its value?

Reply:
In order to make it more clear i made the chainInfo field private with a public getter (so devs can get the current chain info whenever they want). And added the switchChain() method to perform chain hot-swaps